### DISCUSSION TITLE
airflow typeerror cannot use a string pattern on a bytes-like object

### DISCUSSION THREAD

**Discussion by Baft** [apache/airflow#35608] [General] [answered]
### Apache Airflow version

2.7.3

### What happened

install using 

`
virtualenv --python=/usr/bin/python3.8  --prompt="airflow-venv"  /usr/lib/airflow/airflow_venv

source /usr/lib/airflow/airflow_venv/bin/activate 

pip install apache-airflow==2.7.3
`
 run with default settings
`
airflow db init
airflow users create --email admin@example.org --firstname admin --lastname admin -p admin --role Admin --username admin
airflow standalone

also run with production setting on postgres,...

`

got this error

> {app.py:1744} ERROR - Exception on / [GET]
webserver  | Traceback (most recent call last):
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/flask/app.py", line 2529, in wsgi_app
webserver  | response = self.full_dispatch_request()
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/flask/app.py", line 1826, in full_dispatch_request
webserver  | return self.finalize_request(rv)
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/flask/app.py", line 1847, in finalize_request
webserver  | response = self.process_response(response)
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/flask/app.py", line 2344, in process_response
webserver  | self.session_interface.save_session(self, ctx.session, response)
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/airflow/www/session.py", line 33, in save_session
webserver  | return super().save_session(*args, **kwargs)
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/flask_session/sessions.py", line 573, in save_session
webserver  | response.set_cookie(app.config["SESSION_COOKIE_NAME"], session_id,
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/werkzeug/sansio/response.py", line 224, in set_cookie
webserver  | dump_cookie(
webserver  | File "/usr/lib/airflow2/airflow_venv/lib/python3.8/site-packages/werkzeug/http.py", line 1303, in dump_cookie
webserver  | if not _cookie_no_quote_re.fullmatch(value):
webserver  | TypeError: cannot use a string pattern on a bytes-like object

### How to reproduce

install using 

`
virtualenv --python=/usr/bin/python3.8  --prompt="airflow-venv"  /usr/lib/airflow/airflow_venv
source /usr/lib/airflow/airflow_venv/bin/activate 
pip install apache-airflow==2.7.3
`
run with default settings
`
airflow db init
airflow users create --email admin@example.org --firstname admin --lastname admin -p admin --role Admin --username admin
airflow standalone

`

### Operating System

ubuntu 20.04

### Versions of Apache Airflow Providers

2.7.3

### Deployment

Virtualenv installation

### Deployment details

virtual environment created in docker container with ubuntu 20.4

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment, marked as answer]
The only way you can be sure released airflow can be installed is via mechanism where you use constraints. https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

The way you are installing it (and this is explained in the docuement and also in my talk https://www.youtube.com/watch?v=zPjIQjjjyHI which I recommend to walkt it is likely that released version of airflow will not be installable because somoene released a breaking dependency and we cannot do anything about it.

Please double-check and report it back when you see it working with it. and run  pip freeze to compare the dependencies you installed with our constraints - maybe for new version of airlfow we might want to limit something.

Please leat us know how it works when you follow the only supported reproducible way of installing airflow.

Converted it into discussion.

**Reply by Baft** [reply]
Thank you. I tested your solution and the problem was exactly what you mentioned. "....  because somoene released a breaking dependency ..."

**Reply by Baft** [reply]
what i trying to install is as
`
pip install  apache-airflow[aiobotocore,async,cgroups,deprecated_api,github_enterprise,google_auth,kerberos,ldap,leveldb,otel,pandas,password,rabbitmq,sentry,statsd,virtualenv,apache.atlas,apache.beam,apache.cassandra,apache.drill,apache.druid,apache.flink,apache.hdfs,apache.hive,apache.impala,apache.kafka,apache.kylin,apache.livy,apache.pig,apache.pinot,apache.spark,apache.sqoop,apache.webhdfs,atlassian.jira,databricks,dbt.cloud,github,slack,tableau,telegram,arangodb,celery,cncf.kubernetes,daskexecutor,docker,elasticsearch,exasol,github,influxdb,jenkins,mongo,microsoft.mssql,mysql,neo4j,odbc,openfaas,oracle,postgres,presto,redis,samba,singularity,trino,common.sql,ftp,grpc,http,imap,jdbc,microsoft.psrp,microsoft.winrm,openlineage,opensearch,papermill,sftp,smtp,sqlite,ssh,doc,doc_gen]==$AIRFLOW_VERSION docutils pytest-runner Cython==0.28 --constraint "${CONSTRAINT_URL}"
`

then i need to make freeze and download wheel file , to use on offline cluster (air gaped)

**Comment by fharenheit** [comment]
I installed Airflow 2.7.3, but the same error occurs and I cannot log in.

**OS**

CentOS 7.9

**Python**

Python 3.8.13

**Database**

PostgreSQL 12

**Airflow Info**

Airflow 2.7.3

```
[root@airflow airflow]# airflow info

Apache Airflow
version                | 2.7.3                                                               
executor               | CeleryExecutor                                                      
task_logging_handler   | airflow.utils.log.file_task_handler.FileTaskHandler                 
sql_alchemy_conn       | postgresql+psycopg2://airflow_user:airflow_user@10.0.1.81/airflow_db
dags_folder            | /opt/airflow/dags                                                   
plugins_folder         | /opt/airflow/plugins                                                
base_log_folder        | /opt/airflow/logs                                                   
remote_base_log_folder |                                                                     
                                                                                             

System info
OS              | Linux                                                                                                            
architecture    | x86_64                                                                                                           
uname           | uname_result(system='Linux', node='airflow', release='3.10.0-1160.el7.x86_64', version='#1 SMP Mon Oct 19        
                | 16:18:59 UTC 2020', machine='x86_64', processor='x86_64')                                                        
locale          | ('en_US', 'UTF-8')                                                                                               
python_version  | 3.8.13 (default, Aug 16 2022, 12:16:29)  [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]                                  
python_location | /opt/rh/rh-python38/root/usr/bin/python3                                                                         
                                                                                                                                   

Tools info
git             | NOT AVAILABLE                                                      
ssh             | OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017                    
kubectl         | NOT AVAILABLE                                                      
gcloud          | NOT AVAILABLE                                                      
cloud_sql_proxy | NOT AVAILABLE                                                      
mysql           | NOT AVAILABLE                                                      
sqlite3         | 3.7.17 2013-05-20 00:56:22 118a3b35693b134d56ebd780123b7fd6f1497668
psql            | psql (PostgreSQL) 12.11                                            
                                                                                     

Paths info
airflow_home    | /opt/airflow                                                                                                     
system_path     | /opt/rh/rh-python38/root/usr/local/bin:/opt/rh/rh-postgresql12/root/bin:/opt/rh/rh-postgresql12/root/bin:/opt/rh/
                | rh-python38/root/usr/local/bin/:/opt/rh/rh-python38/root/bin:/opt/rh/rh-postgresql96/root/usr/bin:/opt/rh/rh-pyth
                | on38/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin                                        
python_path     | /opt/rh/rh-python38/root/usr/local/bin:/opt/rh/rh-python38/root/usr/lib64/python38.zip:/opt/rh/rh-python38/root/u
                | sr/lib64/python3.8:/opt/rh/rh-python38/root/usr/lib64/python3.8/lib-dynload:/opt/rh/rh-python38/root/usr/local/li
                | b64/python3.8/site-packages:/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages:/opt/rh/rh-python38/ro
                | ot/usr/lib64/python3.8/site-packages:/opt/rh/rh-python38/root/usr/lib/python3.8/site-packages:/opt/airflow/dags:/
                | opt/airflow/config:/opt/airflow/plugins                                                                          
airflow_on_path | True                                                                                                             
                                                                                                                                   

Providers info
apache-airflow-providers-amazon     | 8.10.0
apache-airflow-providers-celery     | 3.4.1 
apache-airflow-providers-common-sql | 1.8.0 
apache-airflow-providers-ftp        | 3.6.0 
apache-airflow-providers-github     | 2.4.0 
apache-airflow-providers-http       | 4.6.0 
apache-airflow-providers-imap       | 3.4.0 
apache-airflow-providers-postgres   | 5.8.0 
apache-airflow-providers-sftp       | 4.7.0 
apache-airflow-providers-sqlite     | 3.5.0 
apache-airflow-providers-ssh        | 3.8.1
``` 

**Error Log**

```
10.8.0.6 - - [18/Nov/2023:12:31:44 +0900] "GET / HTTP/1.1" 500 1539 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
[2023-11-18T12:31:44.190+0900] {app.py:1744} ERROR - Exception on /favicon.ico [GET]
Traceback (most recent call last):
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 1826, in full_dispatch_request
    return self.finalize_request(rv)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 1847, in finalize_request
    response = self.process_response(response)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 2344, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/airflow/www/session.py", line 33, in save_session
    return super().save_session(*args, **kwargs)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/flask_session/sessions.py", line 573, in save_session
    response.set_cookie(app.config["SESSION_COOKIE_NAME"], session_id,
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/sansio/response.py", line 224, in set_cookie
    dump_cookie(
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/http.py", line 1303, in dump_cookie
    if not _cookie_no_quote_re.fullmatch(value):
TypeError: cannot use a string pattern on a bytes-like object
[2023-11-18T12:31:44.194+0900] {app.py:1852} ERROR - Request finalizing failed with an error while handling an error
Traceback (most recent call last):
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 1826, in full_dispatch_request
    return self.finalize_request(rv)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 1847, in finalize_request
    response = self.process_response(response)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 2344, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/airflow/www/session.py", line 33, in save_session
    return super().save_session(*args, **kwargs)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/flask_session/sessions.py", line 573, in save_session
    response.set_cookie(app.config["SESSION_COOKIE_NAME"], session_id,
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/sansio/response.py", line 224, in set_cookie
    dump_cookie(
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/http.py", line 1303, in dump_cookie
    if not _cookie_no_quote_re.fullmatch(value):
TypeError: cannot use a string pattern on a bytes-like object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 1847, in finalize_request
    response = self.process_response(response)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/flask/app.py", line 2344, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/airflow/www/session.py", line 33, in save_session
    return super().save_session(*args, **kwargs)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/flask_session/sessions.py", line 573, in save_session
    response.set_cookie(app.config["SESSION_COOKIE_NAME"], session_id,
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/sansio/response.py", line 224, in set_cookie
    dump_cookie(
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/werkzeug/http.py", line 1303, in dump_cookie
    if not _cookie_no_quote_re.fullmatch(value):
TypeError: cannot use a string pattern on a bytes-like object
10.8.0.6 - - [18/Nov/2023:12:31:44 +0900] "GET /favicon.ico HTTP/1.1" 500 1539 "http://10.0.1.81:8080/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
```

**Reply by potiuk** [reply]
Did you install it the only way we recommend to install as explained in the message before or in some other way ?
